### DISCUSSION TITLE
Window Dialog keeps views loaded inside IRegionManager

### DISCUSSION THREAD

**Discussion by taso131** [PrismLibrary/Prism#2899] [WPF] [answered]
Hello, dear Prism team

We are developing an application, which should be able to open new DialogWindows.
When doing so we want to use prism to navigate (RequestNavigate) the views in the DialogWindow.

When first opening up the dialog, the views get loaded and the IRegionManager loads the corresponding views.
Upon closing the window and opening it again, we get an exception saying the regions are already bound.

Now we can manually unload the views and reload them
But we feel like this isn't the way to go, because we don't want to leave behind in every code-behind an unloading formula, especially when the viewnames can differ.

Is there a way to pass a scoped Region Manager, (we use a singleton) a way down for a specific business case, where we can reuse the
regions without unloading and loading them all the time?
Ideally we want to be able to open up a dialogwindow, which won't get it visual tree broken, so we can use RequestNavigate the way we are used to

I hope my problem is kind of understandable.
For further questions please ask

We'd be more than happy for any answers

**Comment by brianlagunas** [comment, marked as answer]
As you can see, you cannot create multiple instances of the same region. You have two choices, clear the views from the region and remove the region from the region manager, or used scoped regions. When clearing views and removing the region, you don't need code behind in every views, just the main dialog view that contains the regions.

The proper way to handle this scenario is to use scoped regions. You can learn about that here  [Prism Problems & Solutions: Showing Multiple Shells](https://pluralsight.pxf.io/XVxR5)

**Reply by taso131** [reply]
Thanks, this did help me with my problem
Haven't finished quite, yet and will update but I was able to advance
